test/e2e/capi_test.go

```diff
@@ -259,3 +261,76 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 		})
 	}
 })
+
+var _ = Describe("When upgrading a workload cluster [K8s-Upgrade]", func() {
+	var (
+		ctx               = context.TODO()
+		identityNamespace *corev1.Namespace
+	)
+	BeforeEach(func() {
+		Expect(e2eConfig.Variables).To(HaveKey(capi_e2e.CNIPath))
+		rgName := fmt.Sprintf("capz-e2e-%s", util.RandomString(6))
+		Expect(os.Setenv(AzureResourceGroup, rgName)).NotTo(HaveOccurred())
+		Expect(os.Setenv(AzureVNetName, fmt.Sprintf("%s-vnet", rgName))).NotTo(HaveOccurred())
+
+		// Opt into Windows for versions greater than or equal to 1.22
```

Comment:
nit: move this comment down a few lines to be directly above the `WINDOWS_WORKER_MACHINE_COUNT` foo